src/test/java/spoon/test/imports/ImportTest.java

```diff
@@ -403,8 +403,8 @@ public void testNestedAccessPathWithTypedParameter() throws Exception {
 		}
 		CtClass<?> mm = launcher.getFactory().Class().get("spoon.test.imports.testclasses2.AbstractMapBasedMultimap");
 		CtClass<?> mmwli = launcher.getFactory().Class().get("spoon.test.imports.testclasses2.AbstractMapBasedMultimap$WrappedList$WrappedListIterator");
-		assertTrue(mmwli.toString().indexOf("AbstractMapBasedMultimap<K, V>.WrappedList.WrappedIterator")>=0);
-		assertTrue(mm.toString().indexOf("AbstractMapBasedMultimap<K, V>.WrappedList.WrappedIterator")>=0);
+		assertTrue(mmwli.toString().indexOf("AbstractMapBasedMultimap<K, V>.WrappedCollection.WrappedIterator")>=0);
+		assertTrue(mm.toString().indexOf("AbstractMapBasedMultimap<K, V>.WrappedCollection.WrappedIterator")>=0);
```

Comment:
> which means if now you redefine class WrappedIterator in WrappedList it won't be take into account.

We can make a test for that. As long as there is no ambiguous name, the algorithm can use arbitrary access path. But once there are two WrappedIterator instances in different scope, then we will see whether it behaves correctly